File: apps/flutter/lib/models/message.dart
/// 站内消息,字段对齐 server/models/message.go。
class Msg {
  final int id;
  final String type;
  final String title;
  final String content;
  bool read; // 标记已读后本地置 true
  final String createdAt;

  Msg({
    required this.id,
    required this.type,
    required this.title,
    required this.content,
    required this.read,
    required this.createdAt,
  });

  factory Msg.fromJson(Map<String, dynamic> j) => Msg(
        id: j['id'] as int? ?? 0,
        type: j['type'] as String? ?? '',
        title: j['title'] as String? ?? '',
        content: j['content'] as String? ?? '',
        read: j['read'] as bool? ?? false,
        createdAt: j['created_at'] as String? ?? '',
      );

  String get typeText => switch (type) {
        'payment_success' => '支付成功',
        'payment_refunded' => '退款到账',
        'return_confirmed' => '归还确认',
        'breach' => '违约通知',
        'order_cancelled' => '订单取消',
        _ => type,
      };
}

/// 解析 {messages,total,page,unread}。
class MessagePage {
  final List<Msg> messages;
  final int total;
  final int unread;
  MessagePage(this.messages, {required this.total, required this.unread});
}

File: server/models/message.go
// 站内消息模型：支付成功/退款成功/归还确认/违约判定等通知
package models

import "time"

// Message 站内消息
type Message struct {
	Id        int64     `orm:"column(id)" json:"id,string"`
	UserID    int64     `orm:"column(user_id)" json:"user_id,string"`
	Type      string    `orm:"column(type);size(32)" json:"type"`
	Title     string    `orm:"column(title);size(128)" json:"title"`
	Content   string    `orm:"column(content);size(512)" json:"content"`
	Read      bool      `orm:"column(is_read);default(false)" json:"read"`
	CreatedAt time.Time `orm:"column(created_at);auto_now_add;type(datetime)" json:"created_at"`
}

func (m *Message) TableName() string { return "messages" }
